src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowWalker/Components/ResultHelper.cs

```diff
@@ -0,0 +1,114 @@
+    {
+        if (searchControllerResults == null || searchControllerResults.Count == 0)
+        {
+            return new List<WindowWalkerListItem>();
+        }
+
+        List<WindowWalkerListItem> resultsList = new List<WindowWalkerListItem>(searchControllerResults.Count);
+        var addExplorerInfo = searchControllerResults.Any(x =>
+            string.Equals(x.Result.Process.Name, "explorer.exe", StringComparison.OrdinalIgnoreCase) &&
+            x.Result.Process.IsShellProcess);
+
+        // Process each SearchResult to convert it into a Result.
+        // Using parallel processing if the operation is CPU-bound and the list is large.
+        resultsList = searchControllerResults
+            .AsParallel()
+            .Select(x => CreateResultFromSearchResult(x, icon))
+            .ToList();
+
+        if (addExplorerInfo && isKeywordSearch && !SettingsManager.Instance.HideExplorerSettingInfo)
+        {
+            resultsList.Add(GetExplorerInfoResult(infoIcon));
+        }
+
+        return resultsList;
+    }
+
+    /// <summary>
+    /// Creates a Result object from a given SearchResult.
+    /// </summary>
+    /// <param name="searchResult">The SearchResult object to convert.</param>
+    /// <param name="icon">The path to the icon that should be used for the Result.</param>
+    /// <returns>A Result object populated with data from the SearchResult.</returns>
+    private static WindowWalkerListItem CreateResultFromSearchResult(SearchResult searchResult, string icon)
+    {
+        var item = new WindowWalkerListItem(searchResult.Result)
+        {
+            Title = searchResult.Result.Title,
+            Icon = new(icon),
+            Subtitle = GetSubtitle(searchResult.Result),
+        };
+        item.MoreCommands = ContextMenuHelper.GetContextMenuResults(item).ToArray();
+
+        return item;
+    }
+
+    /// <summary>
+    /// Returns the subtitle for a result
+    /// </summary>
+    /// <param name="window">The window properties of the result</param>
+    /// <returns>String with the subtitle</returns>
+    private static string GetSubtitle(Window window)
+    {
+        if (window == null || !(window is Window))
+        {
+            return string.Empty;
+        }
+
+        var subtitleText = Resources.wox_plugin_windowwalker_Running + ": " + window.Process.Name;
+
+        if (SettingsManager.Instance.SubtitleShowPid)
+        {
+            subtitleText += $" ({window.Process.ProcessID})";
+        }
+
+        if (!window.Process.IsResponding)
+        {
+            subtitleText += $" [{Resources.wox_plugin_windowwalker_NotResponding}]";
+        }
+
+        if (SettingsManager.Instance.SubtitleShowDesktopName && WindowWalkerCommandsProvider.VirtualDesktopHelperInstance.GetDesktopCount() > 1)
+        {
+            subtitleText += $" - {Resources.wox_plugin_windowwalker_Desktop}: {window.Desktop.Name}";
```

Comment:
actually all these kinda feel like tags

Reply:
I think it would actually be nice to have each desktop have its own section... but I guess when those get blown away when you search you'd still want to see the desktop tags too. 

![image](https://github.com/user-attachments/assets/31cbc669-c419-42db-a01b-76e28108ac90)

Reply:
I keep not saving the settings while testing because the save button is out of view -- this may become an issue.